internal/cloud/realm/client.go

```diff
@@ -17,16 +18,31 @@ const (
 // Client is a Realm client
 type Client interface {
 	Authenticate(publicAPIKey, privateAPIKey string) (AuthResponse, error)
+	GetUserProfile() (UserProfile, error)
+	FindProjectAppByClientAppID(groupIDs []string, app string) ([]App, error)
 	Status() error
 }
 
 // NewClient creates a new Realm client
 func NewClient(baseURL string) Client {
-	return &client{baseURL}
+	return &client{baseURL: baseURL}
+}
+
+// NewAuthClient creates a new Realm client with a session used for Authorization
+func NewAuthClient(baseURL string, session *Session) Client {
+	// TODO: REALMC-7156 should we return an error here?
+	return &client{baseURL, session}
 }
 
 type client struct {
 	baseURL string
+	session *Session
+}
+
```

Comment:
I added a session to the client object here to allow us to create an 'AuthClient', which can implicitly handle authorization for requests that use an Auth client. I created a separate NewAuthClient function to create a client with a session, and the rest of the methods are shared. 'client.do' adds an Authorization header to a request if the session in the client instance is not nil, which I felt was fine